services/ui-src/src/utils/forms/forms.ts

```diff
@@ -203,3 +204,76 @@ export const flattenFormFields = (formFields: FormField[]): FormField[] => {
   compileFields(formFields);
   return flattenedFields;
 };
+
+// repeatable target population choice list
+const repeatableTargetPopulationChoices = (
+  fields: (FormField | FormLayoutElement)[],
+  choiceList: any
+) => {
+  const createLabel = (field: any) => {
+    if (choiceList.indexOf(field) >= 4) {
+      return `Other: ${field.transitionBenchmarks_targetPopulationName}`;
+    } else {
+      return field.transitionBenchmarks_targetPopulationName;
+    }
+  };
+  const formatTargetPopulations = choiceList?.map((field: any) => {
+    return {
+      checked: false,
+      id: field.id,
+      label: createLabel(field),
+      name: field.transitionBenchmarks_targetPopulationName,
+      value: field.transitionBenchmarks_targetPopulationName,
+    };
+  });
+  // update choices with dynamic target population choices
+  if (fields[1]?.props) {
+    fields[1].props.choices = [];
+    fields[1]?.props?.choices.push(...formatTargetPopulations);
+  }
+};
+
+// returns repeated choice lists - reformatting logic is needed once fields are passed into this function
+export const getRepeatableChoiceLists = (
+  fields: (FormField | FormLayoutElement)[],
+  choiceList: ChoiceList
+) => {
+  // check if fields have target population data
+  const isRepeatableTargetPopulations = fields.filter(
+    (field: FormField | FormLayoutElement) =>
+      field.id.match("targetPopulations")
+  );
+
+  // add more conditional logic for other choice lists
+  if (isRepeatableTargetPopulations.length > 0) {
+    repeatableTargetPopulationChoices(fields, choiceList);
+    return fields;
+  } else {
+    return fields;
+  }
+};
+
+/*
+ * This function resets the 'clear' prop on each field after a ChoiceListField calls
+ * clearUncheckedNestedFields(). Upon re-entering a drawer or modal, the field values will
+ * be correctly hydrated.
+ */
+export const resetClearProp = (fields: (FormField | FormLayoutElement)[]) => {
+  fields.forEach((field: FormField | FormLayoutElement) => {
+    switch (field.type) {
+      case "radio":
+      case "checkbox":
+        field.props?.choices.forEach((childField: FieldChoice) => {
+          if (childField?.children) {
+            resetClearProp(childField.children);
+          }
+        });
+        field.props = { ...field.props, clear: false };
+        resetClearProp(field.props?.choices);
+        break;
+      default:
+        field.props = { ...field.props, clear: false };
+        break;
+    }
+  });
+};
```

Comment:
I wanted to learn more about this so I looked at the companion method: `ChoiceListField` -> `clearUncheckedNestedFields`

At the end of each case there I added this line and it seemed to work the same as this additional function. Is there something timing related that would make that an issue?
`child.props = { ...child.props, clear: false };`



```
        switch (child.type) {
            case "radio":
            case "checkbox":
              if (child.props?.choices) {
                child.props.choices.forEach((choice: FieldChoice) => {
                  choice.checked = false;
                });
                child.props = { ...child.props, clear: true };
                form.setValue(child.id, []);
                form.unregister(child.id);
                clearUncheckedNestedFields(child.props.choices);
+               child.props = { ...child.props, clear: false };
              }
              break;
            default:
              child.props = { ...child.props, clear: true };
              form.setValue(child.id, "");
              form.unregister(child.id);
+             child.props = { ...child.props, clear: false };
              break;
          }
```

Reply:
Did this also clear out the nested user input when going from `yes` to `no` and back to `yes`?

Reply:
Holy.. If that works because the `form.setValue()` call is causing a re-render of the fields and thus is letting you call child.props = {...child.props, clear: false }; just after... I'd be so happy

Reply:
Think it might not cover the case though that Britt just mentioned

Reply:
@britt-mo if you have not saved anything yet, and you select `yes`, fill all the fields out, select `no`, then select `yes` again, it does clear all the fields.
However if you select `yes`, fill out the fields, save it, re-enter, then `no` then `yes` it does _not_ clear them. So I think my suggestion doesn't cover the case you specified.
That said, why would `clearUncheckedNestedFields` not handle that 🤔 
I'm ok with approving now that I see the benefit this provides. Just poking at it to see where the real breakdown is.

Reply:
@gmrabian I'm sure there's a cleaner way to get this working lol but at the same time I know all this code will get removed once we're using Zustand so I'm inclined to move forward with this bandaid solution.